frontend/packages/console-shared/src/components/status/PopoverStatus.tsx

```diff
@@ -2,16 +2,38 @@ import * as React from 'react';
 import { Button, Popover, PopoverPosition } from '@patternfly/react-core';
 import StatusIconAndText from './StatusIconAndText';
 
-const PopoverStatus: React.FC<React.ComponentProps<typeof StatusIconAndText>> = ({
+type PopoverStatusProps = React.ComponentProps<typeof StatusIconAndText> & {
+  activeIcon?: React.ReactElement;
+  hideHeader?: boolean;
+};
+
+const PopoverStatus: React.FC<PopoverStatusProps> = ({
   title,
+  hideHeader,
+  icon,
+  activeIcon,
   children,
   ...other
-}) => (
-  <Popover position={PopoverPosition.right} headerContent={title} bodyContent={children}>
-    <Button variant="link" isInline>
-      <StatusIconAndText {...other} title={title} />
-    </Button>
-  </Popover>
-);
+}) => {
+  const [isActive, setActive] = React.useState(false);
+  return (
+    <Popover
+      position={PopoverPosition.right}
+      headerContent={hideHeader ? null : title}
+      bodyContent={children}
+      aria-label={title}
+      onHide={React.useCallback(() => setActive(false), [])}
```

Comment:
I think we should extract these memoized callbacks into separate variables.

Also, any function used within these callbacks should be treated as their dependency.

```ts
const [isActive, setActive] = React.useState(false);
const onHide = React.useCallback(() => setActive(false), [setActive]);
```

Reply:
fixed